R/get_baseline_predictions.R

```diff
@@ -0,0 +1,136 @@
+#' Get predictions for a single baseline model
+#'
+#' @param target_ts a `data.frame` of target data in a time series format
+#'   (contains columns `time_index`, `location`, and `observation`) for a single
+#'   location
+#' @param transformation string specifying the transformation used on the
+#'   distribution which determines its shape; can be one of "none" or "sqrt".
+#' @param symmetrize boolean specifying whether to make the distribution symmetric;
+#'   can be one of `TRUE` or `FALSE`.
+#' @param window_size integer specifying how many previous observations in the
+#'   target data should be used to inform the forecasts
+#' @param effective_horizons numeric vector of prediction horizons relative to
+#'   the last observed date in `target_ts`
+#' @param origin string specifying the origin to use when making predictions;
+#'   recommended to be "median" if the temporal resolution is daily and "obs"
+#'   if weekly or otherwise. Defaults to "obs".
+#' @param quantile_levels numeric vector of quantile levels to output; set to NULL
+#'   if quantile outputs not requested. Defaults to NULL.
+#' @param n_samples integer of amount of samples to output (and predict);
+#'   set to NULL if sample outputs not requested (in this case 100000 samples
+#'   are generated from which to extract quantiles). Defaults to NULL.
+#' @param round_predictions boolean specifying whether to round the output
+#'   predictions to the nearest whole number. Defaults to FALSE
+#' @param seed integer specifying a seed to set for reproducible results.
+#'   Defaults to NULL, in which case no seed is set.
+#'
+#' @return data frame of a baseline forecast for one location, one model with
+#'   columns `horizon` , `output_type`, `output_type_id`, and `value`,
+#'   but which are stored as a nested list in a 1x1 data frame
+#'
+#' @importFrom rlang .data
+
+get_baseline_predictions <- function(target_ts,
+                                     transformation,
+                                     symmetrize,
+                                     window_size,
+                                     effective_horizons,
+                                     origin = "obs",
+                                     quantile_levels = NULL,
+                                     n_samples = NULL,
+                                     round_predictions = FALSE,
+                                     seed = NULL) {
+  # validate arguments
+  validate_target_ts(target_ts)
+
+  num_locs <- length(unique(target_ts[["location"]]))
+  if (num_locs != 1) {
+    cli::cli_abort("{.arg target_ts} contains {.val num_locs} but only one may be provided.")
+  }
```

Comment:
I wrote the code like this based on the hubverse framework of target data in a time series format always having the three columns "time_index", "location", and "observation", so the validation function for target time series data also makes this assumption. I don't know if this framework has changed for the hubverse, or if we'd prefer to do something different for this package, but I think we should discuss and then decide on a package-level, rather than just for this function in particular.